packages/shader-lab/src/codeGen/GLES300.ts

```diff
@@ -129,4 +126,27 @@ export class GLES300Visitor extends GLESVisitor {
     }
     return super.visitVariableIdentifier(node);
   }
+
+  override visitJumpStatement(node: ASTNode.JumpStatement): string {
+    if (node.isFragReturnStatement) {
+      const { _referencedVaryingList, mrtStruct } = VisitorContext.context;
+      if (mrtStruct) {
+        return "";
+      }
+      if (!_referencedVaryingList[V3_GL_FragColor]) {
+        const token = Token.pool.get();
+        token.set(ETokenType.ID, V3_GL_FragColor, ShaderLab.createPosition(0, 0, 0));
+        _referencedVaryingList[V3_GL_FragColor] = {
+          ident: token,
+          typeInfo: new SymbolType(EKeyword.VEC4, "vec4"),
+          qualifier: "out",
+          astNode: node
+        };
+      }
```

Comment:
Encapsulate this logic that repeats `visitVariableIdentifier` into a function, such as `registerFragColorVariable`

Reply:
done